backend/src/routes/MyRestaurantRoute.ts

```diff
@@ -0,0 +1,27 @@
+import express from 'express';
+import multer from 'multer';
+import MyRestaurantController from '../controllers/MyRestaurantController';
+import { jwtCheck, jwtParse } from '../middleware/auth';
+import { validateMyRestaurantRequest } from '../middleware/validation';
+
+const router = express.Router();
+
+const storage = multer.memoryStorage();
+
+const upload = multer({
+  storage,
+  limits: {
+    fileSize: 5 * 1024 * 1024,
+  },
+});
+
+router.post(
+  '/',
+  upload.single('imageFile'),
```

Comment:
<div>

<strong>Bito Code Review Agent Run</strong> <a href=https://github.com/omkardongre/Food-Ordering-App/pull/27#issuecomment-2383391092>#6fe9f8</a><span> - 09/30/2024, 02:40 pm</span>
<br>

<div><small>🔴</small> High importance</div>

<div id="suggestion">
<div id="issue">
<b>Issue:</b> The order of middleware in the POST route may not be optimal for security and efficiency
</div>
<div id="fix">
<b>Fix:</b> Reorder the middleware functions in the POST route to perform authentication first, then validation, and finally file upload
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -18,9 +18,9 @@
  router.post(
    '/',
 -  upload.single('imageFile'),
 -  validateMyRestaurantRequest,
    jwtCheck,
    jwtParse,
 +  validateMyRestaurantRequest,
 +  upload.single('imageFile'),
    MyRestaurantController.createMyRestaurant,
  );
```
</div>
</details>
</div>
<br>


</div>

Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged